src/transformStateWithClones.js

```diff
@@ -6,8 +6,27 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let stateCopy = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      stateCopy = {};
+    } else if (action.type === 'addProperties') {
+      stateCopy = { ...stateCopy, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      stateCopy = { ...stateCopy };
+
+      action.keysToRemove.forEach((key) => {
+        delete stateCopy[key];
+      });
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. This will make the code more readable and organized. Replace the if-else chain with a switch statement.